integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/RddExecutionContext.java

```diff
@@ -235,6 +238,11 @@ protected ZonedDateTime toZonedTime(long time) {
                     runId, sparkContext.getParentJobName(), sparkContext.getJobNamespace()));
   }
 
+  protected SparkVersionFacet buildSparkVersionFacet() {
+    String jarVersion = this.getClass().getPackage().getImplementationVersion();
+    return new SparkVersionFacet(SparkSession.active().version(), jarVersion);
```

Comment:
I'd move the logic of how to obtain these fields into the `SparkVersionFacet` itself. Maybe just pass in the `SparkSession` as its constructor arg.